test-integration/filter-ens.js

```diff
@@ -0,0 +1,54 @@
+/* global artifacts */
+
+const ethers = require("ethers");
+const utils = require("../utils/utilities.js");
+const { deployArgent } = require("../utils/argent-deployer.js");
+
+const ENSRegistry = artifacts.require("ENSRegistry");
+const ArgentENSManager = artifacts.require("ArgentENSManager");
+const ArgentENSResolver = artifacts.require("ArgentENSResolver");
+const ArgentEnsManagerFilter = artifacts.require("ArgentEnsManagerFilter");
+
+const ARGENT_ENS_ADDRESS = "0xF32FDDEF964b98b1d2d2b1C071ac60ED55d4D217";
+const ARGENT_ENS_OWNER_ADDRESS = "0xa5c603e1C27a96171487aea0649b01c56248d2e8";
+
+contract("ENS Filter", (accounts) => {
+  let argent;
+  let ensRegistry;
+  let ensResolver;
+  let argentEnsManager;
+
+  before(async () => {
+    argent = await deployArgent(accounts);
+
+    argentEnsManager = await ArgentENSManager.at(ARGENT_ENS_ADDRESS);
+    await argentEnsManager.addManager(argent.infrastructure, { from: ARGENT_ENS_OWNER_ADDRESS });
+
+    ensRegistry = await ENSRegistry.at(await argentEnsManager.ensRegistry());
+    ensResolver = await ArgentENSResolver.at(await argentEnsManager.ensResolver());
+
+    const filter = await ArgentEnsManagerFilter.new();
+    await argent.dappRegistry.addDapp(0, argentEnsManager.address, filter.address);
+  });
+
+  it("should register an ENS name", async () => {
```

Comment:
This tests that a manager of the `ArgentENSManager` can register the ENS of a wallet. That test should be in `test/ens.js` .

However, in `test-integration` we should test that a wallet can register his ENS via a multicall and the dapp registry. Something that is currently in `test/ens.js` under the `ENS Integrations` section...

Reply:
Pushed new changes